pyanaconda/modules/storage/bootloader/base.py

```diff
@@ -116,24 +117,31 @@ def partition_p(arg):
         for nic in config:
             ip_params.add("%s:%s" % (nic, ",".join(sorted(config[nic]))))
 
-        # update the set
-        self.clear()
-        self.update(rest)
-        self.update(ip_params)
-        return self
+        # update the list
+        self._arguments.clear()
+        self._arguments.extend(rest)
+        self._arguments.extend(ip_params)
+        return self._arguments
 
     def __str__(self):
+        """Convert the argument list to string.
+
+        Merges ip= arguments targeting the same interface as a side effect.
+        """
         self._merge_ip()
-        return " ".join(list(self))
+        return " ".join(list(self._arguments))
 
-    def add(self, key):
-        self.discard(key)
-        super().add(key)
```

Comment:
Please leave the `add` there. It's not nice to making list from everything. Or better add new `extend` method which will work with lists and change the `update` method.

Reply:
I think there are `add` and `update` method, because the class behaves like an ordered set. I think we shouldn't change that. The list is only an implementation detail.